moz-webgpu-cts/src/main.rs

```diff
@@ -1347,6 +1348,109 @@ fn run(cli: Cli) -> ExitCode {
             println!("Full analysis: {analysis:#?}");
             ExitCode::SUCCESS
         }
+        Subcommand::UpdateBacklog => {
+            let mut files = {
+                let mut found_parse_err = false;
+                let extracted = read_and_parse_all_metadata(&gecko_checkout)
+                    .filter_map(|res| match res {
+                        Ok(ok) => Some(ok),
+                        Err(AlreadyReportedToCommandline) => {
+                            found_parse_err = true;
+                            None
+                        }
+                    })
+                    .collect::<BTreeMap<_, _>>();
+                if found_parse_err {
+                    log::error!(concat!(
+                        "found one or more failures while parsing metadata, ",
+                        "see above for more details"
+                    ));
+                    return ExitCode::FAILURE;
+                }
+                extracted
+            };
+
+            #[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+            enum Case {
+                #[default]
+                PermaPass,
+                PermaFail,
+                Other,
+            }
+            let mut found_write_err = false;
+            for (file_path, file) in files.iter_mut() {
+                let File {
+                    properties: _,
+                    tests,
+                } = file;
+
+                for (_name, test) in tests {
+                    // let SectionHeader(name) = &name;
+                    // let test_path = TestPath::from_fx_metadata_test(
+                    //     file_path.strip_prefix(gecko_checkout).unwrap(),
+                    //     name,
+                    // )
+                    // .unwrap();
+                    let Test {
+                        properties,
+                        subtests,
+                    } = test;
+                    let mut cases = ExpandedPropertyValue::default();
+                    for ((platform, build_profile), expected) in properties
+                        .expected
+                        .as_ref()
+                        .unwrap_or(&Default::default())
+                        .iter()
+                    {
+                        let case = match expected.as_permanent() {
+                            Some(TestOutcome::Ok) => Case::PermaPass,
+                            _ => Case::Other,
+                        };
+                        cases[(platform, build_profile)] = case;
+                    }
+                    if !subtests.is_empty() {
```

Comment:
issue: Got a bug here where expected test outcomes are ignored when subtests are present. We should take _both_ into account.